src/primitives/ray.jl

```diff
@@ -42,3 +42,13 @@ origin(r::Ray) = r.p
 The direction of the ray.
 """
 direction(r::Ray) = r.v
+
+function Base.in(p::Point, r::Ray)
+	w = norm(r.v)
+	d = evaluate(Euclidean(), p, Line(r(0), r(1)))
+	if d+w ≈ w
+		return sum((p - r.p) .* r.v) >= 0 # return true if point is in ray
+	else
+		return false
+	end
+end
```

Comment:
```suggestion
```

This function is out of the scope of this PR. We can add it in a separate PR after we fix the bugs in the intersection you mentioned. The more self-contained the better. In the future we can come back to this PR and understand the fix.